pkg/pillar/scripts/device-steps.sh

```diff
@@ -209,6 +184,41 @@ reload_ntp_sources() {
     populate_ntp_sources "$ns"
 }
 
+# Poll if NTP servers and reload if changed
+reload_ntp_sources_in_loop() {
+    while true; do
+        ns=$(get_ntp_servers)
+        if [ -n "$ns" ] && [ "$ns" != "$NTPSERVERS" ]; then
+            echo "$(date -Ins -u) NTP server changed from \"$NTPSERVERS\" to \"$ns\", reload NTP sources"
+            # Reload NTP sources
+            reload_ntp_sources "$ns"
+        fi
+        # Why strange number? Easy to spot in `ps` output and debug.
+        sleep 30.1
+    done
+}
+
+# Start NTP daemon
+start_ntp_daemon() {
+    echo "$(date -Ins -u) Check for NTP config"
+    if [ -f /usr/sbin/chronyd ]; then
+        # Run chronyd to keep it in sync.
+        echo "$(date -Ins -u) chronyd -u root -f /etc/chrony/chrony.conf"
+        /usr/sbin/chronyd -u root -f /etc/chrony/chrony.conf
+        ret_code=$?
+        echo "$(date -Ins -u) chronyd: $ret_code"
+        # Add chronyd to watchdog
+        touch "$WATCHDOG_PID/chronyd.pid"
```

Comment:
Do you add it to watchdog even if ret_code was non-zero? In that case it wouldn't be running, right?

Reply:
Frankly this is a grey area and not clear what we do if something fails from the very beginning (we have bunch of code like that). If chronyd unexpectedly dies - we reboot. No doubts. But if chronyd does not start? Seems also reboot is a good idea.

Reply:
I guess we have (or have had) code in device-steps.sh which calls exit when unexpected things happen, which has the same effect. But perhaps we should make them more explicit by logging "chronyd failed to start - rebooting?' and then call reboot? Or retry 3 times before that?